src/utils/authentication/authentication.service.ts

```diff
@@ -1,64 +1,85 @@
-import {
-  FakeDB,
-  type AuthMethod,
-  type DbAuthenticateData,
-  type DbUser
-} from "~utils/authentication/fakeDB";
+import type { AuthProviderType } from "embed-api";
+import { supabase, trpcVanilla } from "~utils/embedded/embedded.utils";
+import type { Provider } from "@supabase/supabase-js";
 
-async function refreshSession(): Promise<DbAuthenticateData | null> {
-  return FakeDB.refreshSession();
-}
+const SUPABASE_PROVIDER_BY_AUTH_PROVIDER_TYPE: Record<
+  AuthProviderType,
+  Provider | null
+> = {
+  PASSKEYS: null,
+  EMAIL_N_PASSWORD: null,
+  GOOGLE: "google",
+  FACEBOOK: "facebook",
+  X: "twitter",
+  APPLE: "apple"
+};
 
-async function authenticate(
-  authMethod: AuthMethod
-): Promise<DbAuthenticateData | null> {
-  return FakeDB.authenticate(authMethod);
-}
+async function authenticate(authProviderType: AuthProviderType) {
```

Comment:
Once we have the new screens for email and password, we probably want to add an `emailSignUp` and `emailSignIn` to this service and update this `authenticate` function.